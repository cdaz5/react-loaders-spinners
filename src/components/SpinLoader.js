import styled, { keyframes } from 'styled-components';
import { string, number, bool } from 'prop-types';

const spin = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;

const SpinLoader = styled.div`
	border-radius: 50%;
	border: ${props =>
    props.thickness && props.pColor
      ? `${props.thickness}px solid ${props.pColor}`
      : props.thickness
        ? `${props.thickness}px solid #f3f3f3`
        : props.pColor
          ? `10px solid ${props.pColor}`
          : '10px solid #f3f3f3'};
	border-top: ${props =>
    props.thickness && props.sColor
      ? `${props.thickness}px solid ${props.sColor}`
      : props.thickness
        ? `${props.thickness}px solid #555`
        : props.sColor
          ? `10px solid ${props.sColor}`
          : '10px solid #555'};
	height: ${props => (props.height ? `${props.height}px` : '50px')};
	width: ${props => (props.width ? `${props.width}px` : '50px')};
	animation: ${spin} 1s linear infinite;
`;

SpinLoader.propTypes = {
  width: number,
  height: number,
  thickness: number,
  loading: bool.isRequired,
  pColor: string,
  sColor: string
};

SpinLoader.defaultProps = {
  width: 50,
  height: 50,
  thickness: 10,
  pColor: '#f3f3f3',
  sColor: '#555',
  loading: true
};

export default SpinLoader;
