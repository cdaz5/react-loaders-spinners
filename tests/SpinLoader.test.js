import React from 'react';
import renderer from 'react-test-renderer';

import SpinLoader from '../src/components/SpinLoader';

describe('SpinLoader', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<SpinLoader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
