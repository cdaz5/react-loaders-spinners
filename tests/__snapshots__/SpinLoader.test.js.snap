// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpinLoader matches snapshot 1`] = `
<div
  className="sc-bdVaJa cIUElR"
  height={50}
  width={50}
/>
`;
